core/src/main/kotlin/io/github/nefilim/kjwt/ClaimValidation.kt

```diff
@@ -77,21 +75,23 @@ object ClaimsVerification {
         error: KJWTValidationError = KJWTValidationError.RequiredClaimIsInvalid(name),
     ): ClaimsValidator = { claims ->
         when (claims.claim()) {
-            is Some -> predicate((claims.claim() as Some<T>).value).toValidatedNel(error, claims)
-            is None -> claims.validNel()
+            is Some -> predicate((claims.claim() as Some<T>).value).toEitherNel(error, claims)
+            is None -> claims.right()
         }
     }
 
-    private fun Boolean.toValidatedNel(invalid: KJWTValidationError, claims: JWTClaims): ClaimsValidatorResult {
+    private fun Boolean.toEitherNel(invalid: KJWTValidationError, claims: JWTClaims): ClaimsValidatorResult {
         return if (this)
-            claims.validNel()
+            claims.right()
         else
-            invalid.invalidNel()
+            invalid.leftNel()
     }
 
+
     fun validateClaims(vararg validations: ClaimsValidator): ClaimsValidator = { claims ->
-        validations.fold(Validated.validNel(claims)) { c, t ->
-            c.zip(Semigroup.nonEmptyList(), t(claims)) { c1, _ -> c1 }
-        }
+        validations.asSequence().mapOrAccumulate(
+            combine = { e1, e2 -> e1 + e2 },
+            transform = { it(claims).bind() })
+            .map { it.last() }
```

Comment:
@Atternatt will this gather all the errors instead of short circuit at the first error instance?

Reply:
yes this is the same as the old implementation you did but in 2.0.0 Higher kinds disappeared. mayor accumulate will accumulate errors before shifting. I had to implement the combine function in order to join all the listnel errors

Reply:
the it.last() does mostly the same as your initial implementation. if I'm not mistaken at the end the object pass d is the same right? like a composite